Gems/DebugDraw/Assets/Shaders/ObbIntersection.azsl

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/Features/RayTracing/RayTracingIntersectionAttributes.azsli>
+#include <Atom/Features/SrgSemantics.azsli>
+
+// https://github.com/erich666/GraphicsGems/blob/master/gems/RayBox.c
+bool IntersectRayAABB(float3 rayOrigin, float3 rayDirection, float3 aabbMin, float3 aabbMax, out float t, out float3 normal)
+{
+    t = 0.f;
+    normal = float3(1, 0, 0);
+
+    float3 maxT;
+
+    // TODO: Do we need to detect if the ray starts inside of the AABB
+
```

Comment:
Fix(ed) this?!

Reply:
Removed comment; not handling being inside of the AABB gives it an equivalent look to rasterizing the surface, which makes sense.